### PR TITLE
feat: add initial resource database

### PR DESCRIPTION
**Pull request by coryan**
testbench.database will hold all the resources (Bucket, Object, etc.) so
the Flask and gRPC applications can find them and implement the various
RPCs associated with them.

Part of the work for #24

### CODE REVIEW

**Comment by coryan** [comment]
Ping

File: testbench/database.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            metageneration, match, not_match, True, context
+        )
+
+    def get_bucket_without_generation(self, bucket_name, context):
+        bucket = self.buckets.get(bucket_name)
+        if bucket is None:
+            testbench.error.notfound("Bucket %s" % bucket_name, context)
+        return bucket
+
+    def insert_bucket(self, request, bucket, context):
```
Comment (cojenco) [inline testbench/database.py:64]: How can `metageneration` be applied to the bucket when calling `insert_bucket`? Comparing this to `insert_test_bucket` where on line 104
```
bucket_test.metadata.metageneration = 4
bucket_test.metadata.versioning.enabled = True
```

Context: When creating a new bucket against the emulator (ultimately calling `insert_bucket`, the new bucket does not  have a `metageneration` attached. Is this WAI? If so, how can one add a `metageneration` to a newly created bucket?

**Review by cojenco — COMMENTED** [review]
A question on `insert_bucket`. Otherwise, LGTM, thanks!

File: testbench/database.py (hunk 1 above)
Comment (coryan) [reply testbench/database.py:64]: In the old repo we have this:

https://github.com/googleapis/google-cloud-cpp/blob/f9d58d68ed14d00c583cecd6381fcd6e45c9bae1/google/cloud/storage/emulator/gcs/bucket.py#L209

In this repo we have this:

https://github.com/googleapis/storage-testbench/blob/main/gcs/bucket.py#L211

I think it is a bug to initialize metageneration to `0`, it should be `1`.  That is probably why you see "no" metageneration.

In the old repo we also have this:


https://github.com/googleapis/google-cloud-cpp/blob/f9d58d68ed14d00c583cecd6381fcd6e45c9bae1/google/cloud/storage/emulator/gcs/bucket.py#L273-L274

that is also a bug, the metageneration should be incremented in all changes, regardless of the versioning status.  I changed that already:

https://github.com/googleapis/storage-testbench/blob/85e9d6d39d453ac007455e27aafc58045956293e/gcs/bucket.py#L275

In short.  I think `insert_bucket` *should*, but probably does not, create buckets with a valid metageneration, but that is not a problem in the `database.py` file.

Comment (cojenco) [reply testbench/database.py:64]: I see. Thanks for the clear explanation and fix.

**Review by cojenco — APPROVED** [review]
